crates/util/src/proxy.rs

```diff
@@ -0,0 +1,108 @@
+#[cfg(target_os = "macos")]
+use system_configuration::{
+    core_foundation::{
+        base::CFType,
+        dictionary::CFDictionary,
+        number::CFNumber,
+        string::{CFString, CFStringRef},
+    },
+    dynamic_store::SCDynamicStoreBuilder,
+    sys::schema_definitions::{
+        kSCPropNetProxiesHTTPEnable, kSCPropNetProxiesHTTPPort, kSCPropNetProxiesHTTPProxy,
+        kSCPropNetProxiesHTTPSEnable, kSCPropNetProxiesHTTPSPort, kSCPropNetProxiesHTTPSProxy,
+        kSCPropNetProxiesSOCKSEnable, kSCPropNetProxiesSOCKSPort, kSCPropNetProxiesSOCKSProxy,
+    },
+};
+
+pub fn http_proxy_from_env() -> Option<isahc::http::Uri> {
+    let all_keys = [
+        "ALL_PROXY",
+        "all_proxy",
+        "HTTPS_PROXY",
+        "https_proxy",
+        "HTTP_PROXY",
+        "http_proxy",
+    ];
+    for env_key in all_keys {
+        if let Ok(env) = std::env::var(env_key) {
+            return env.parse::<isahc::http::Uri>().ok();
+        }
+    }
+    proxy_from_platform()
+}
+
+#[cfg(target_os = "macos")]
+fn proxy_from_platform() -> Option<isahc::http::Uri> {
```

Comment:
@Congyuwang Here, the system proxy can be accessed. The entire process prioritizes reading Zed's proxy settings first, followed by the system proxy.

Reply:
That’s great. But I’m also thinking what about putting all these directly as settings, like some other softwares do, give proxies several options:

- Follow system proxy / automatic
- Off
- Manual

I see that a lot of apps seem to arrange proxy settings like this.